### ISSUE TITLE
Add support for privileged users

### ISSUE THREAD

**Issue by jpdillingham** [jpdillingham/Soulseek.NET#209]
This includes:

* [x] Handling of the initial privileged user list sent at login (PrivilegedUsers/69)
* [x] Handling of new privileged user notifications (AddPrivilegedUser/91)
* [x] Ability to check your own privileged status (CheckPrivileges/92)
* [x] Ability to check another user's privileged status (UserPrivileges/122)
* [x] Handling of notification of privilege grants (NotifyPrivileges/124)
* [x] Ability to acknowledge (as well as automatically, as with private messages) notification of privilege grants (AcknowledgeNotifyPrivileges/125)
* [x] Ability to grant privileges to users (GivePrivileges/123)

**Comment by jpdillingham** [comment]
Notification messages `AddPrivilegedUser` and `NotifyPrivileges` are highly ambiguous; it isn't clear if `NotifyPrivileges` supercedes the former, but it seems likely.

`NotifyPrivileges` includes a username, which would imply that this notification could be sent with regards to users other than the currently logged in user.

Until I have more information about the differences here, other than the fact that `NotifyPrivileges` expects to be acknowledged with `AcknowledgeNotifyPrivileges` and the former does not, I'm going to use the same event for both with a nullable integer for the notification id.

Implementing code will need to examine the id and send an acknowledgement if it is non-null.

**Comment by jpdillingham** [comment]
The `GivePrivileges` message has no direct response message, and as far as I can tell there's no positive indication anywhere that the command was successful.

The command may fail if the target user doesn't exist, if you don't have any privileges, or if the number of specified privileges exceeds the number of privileged days currently granted to you.  The failure message comes by way of a private message from `server` with `IsAdmin = true`.

I've made the choice to design `GrantUserPrivilegesAsync()` to be completely agnostic about the outcome of the command, and instead implementing code will need to monitor private messages to trap and report errors.  I've tried to keep the library as close to the Soulseek protocol as possible up to this point, and this decision is in alignment with that goal.
